### PR TITLE
fs-1882-user-roles-error-page

### PR DESCRIPTION
**Pull request by gio-karunakaran**

### CODE REVIEW

File: frontend/user/templates/user.html [hunk 1]
```diff
@@ -3,33 +3,51 @@
...
+
+    {% if roles_error %}
+    <p class="govuk-body">You do not have access
+    <p class="govuk-body">Either you:</p>
+    <ul class="govuk-list govuk-list--bullet">
+        <li class="govuk-body">
+            do not have the right account permissions set up, or
+        </li>
+        <li class="govuk-body">
+            the link you’re using has expired
```
Comment (harryyo) [inline frontend/user/templates/user.html:19]: Did we ever discover why this text is here? As no-one will be here from an expired link?

File: frontend/user/routes.py [hunk 2]
```diff
@@ -15,10 +16,22 @@
 @user_bp.route("")
 @login_requested
 def user():
```
Comment (harryyo) [inline frontend/user/routes.py:18]: May be good to have a docstring in here now that it's a bit dynamic?

File: frontend/user/routes.py [hunk 3]
```diff
@@ -15,10 +16,22 @@
...
+    roles_required = request.args.get("roles_required")
     logged_in_user = g.user if g.is_authenticated else None
+    roles_error = False
+    if logged_in_user and (
+        not logged_in_user.roles
```
Comment (harryyo) [inline frontend/user/routes.py:23]: I wonder, if we only show a roles_required error when the user is logged in, perhaps all the roles required logic should go inside an `if logged_in_user:` block?

Comment (harryyo) [reply frontend/user/routes.py:23]: we could then just lose the roles_error flag and instead just validate to check that the roles_required are actually required eg.
```suggestion
        if logged_in_user:
             if roles_required and logged_in_user.roles 
             and all(role_required in logged_in_user.roles
            for role_required in roles_required):
                 roles_required = None
```
...and then just pass roles_required to the template and build the logic off that...? Any thoughts?

**Review by harryyo — COMMENTED** [review]
Great, just thoughts for chat

File: frontend/user/routes.py (hunk 2 above)
Comment (gio-karunakaran) [reply frontend/user/routes.py:18]: added docstrings! :D

File: frontend/user/templates/user.html (hunk 1 above)
Comment (gio-karunakaran) [reply frontend/user/templates/user.html:19]: edited now after confirming with peter 👍🏾

File: frontend/user/routes.py (hunk 3 above)
Comment (gio-karunakaran) [reply frontend/user/routes.py]: edited with required changes

**Review by harryyo — APPROVED** [review]
Sweet

**Pull request merged**